src/App.tsx

```diff
@@ -1,34 +1,77 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import './App.css'
+import './App.css';
+import Navbar from './components/Navbar';
+import Categories from './components/Categories';
+import ProductPage from './components/ProductPage';
+import Cart from './components/Cart';
+import { useState } from 'react';
+
+interface ProductCartItem {
+  id: string;
+  image: string;
+  title: string;
+  price: number;
+  category: string;
+  description: string;
+  quantity: number;
+}
 
 function App() {
-  const [count, setCount] = useState(0)
+  // gett current route
+  const currentPath = window.location.pathname;
+
+  // state variables
+  const [isCartOpen, setIsCartOpen] = useState(false);
+  const [cartProducts, setCartProducts] = useState<ProductCartItem[]>([]);
+  const [cartCount, setCartCount] = useState(0);
+
+  // cart handling
+  const toggleCart = () => {
+    console.log(isCartOpen);
+    setIsCartOpen(!isCartOpen);
+  };
+
+  const handleAddToCart = (newProduct: ProductCartItem) => {
+    setCartCount(cartCount + 1);
+
+    setCartProducts((prevProducts) => {
+      // if product in array matches product we are trying to add - remember it
+      const existingProduct = prevProducts.find((p) => p.id === newProduct.id);
+
+      if(existingProduct) {
+        // we have a product that is already in array (dup item) - return every other array item as normal but return dup item with incremented quantity
+        return prevProducts.map((p) =>
+          p.id === newProduct.id ? { ...p, quantity: p.quantity + 1 } : p
+        );
+      }
+      // if product is new, add to end of list with quantity of 1
+      return [...prevProducts, { ...newProduct, quantity: 1}];
+    });
+  }
```

Comment:
For application code, especially in a large project we've tried to develop a style that's as declarative as possible. When possible avoid declaring intermediate variables like `existingProduct`. A more declarative style might look like:
```suggestion
    setCartProducts((prevProducts) =>
      prevProducts.some((p) => p.id === newProduct.id)
        ? // we have a product that is already in array (dup item) - return every other array item as normal but return dup item with incremented quantity
          prevProducts.map((p) =>
            p.id === newProduct.id ? { ...p, quantity: p.quantity + 1 } : p
          )
        : // if product is new, add to end of list with quantity of 1
          [...prevProducts, { ...newProduct, quantity: 1 }]
    )
  }
```